pine64/pinebook-pro/default.nix

```diff
@@ -8,6 +8,10 @@
 
   boot.kernelPackages = lib.mkDefault pkgs.linuxPackages_latest;
 
+  boot.kernelParams = [
+    "console=tty0"
```

Comment:
@samueldr what is your opinion on this?

Related: https://github.com/NixOS/nixos-hardware/pull/745#discussion_r1341922235

Reply:
I guess by default it will try some serial device instead?

Reply:
I have some understanding of the option in case you have questions.

Reply:
I think I misspoke in the previous comment in what I think the best option is for this.

A default might be right, and it would depend on the device type.

We can't rely on the "unconfigured default", since in DT-land Linux defaults it relies on `/chosen/stdout-path`, which sometimes doesn't make sense given a target:

https://github.com/torvalds/linux/blob/1c8b86a3799f7e5be903c3f49fcdaee29fd385b5/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts#L28-L30

But setting `console=` without qualifying it is kinda rude as it prevents the user from "simply" choosing another one.

So the workaround (with our semantics) is to put it at the leftmost location. In other words:

```
boot.kernelParams = mkBefore [ "console=tty0" ];
```

This would be for devices we can ascertain that most users would expect a given default console choice.